### PR TITLE
Support timers in realm

### LINKED ISSUE #205: setTimeout & setInterval not working
Problem:
**setTimeout** & **setInterval** can't be used on the visualizer right now. It gives the error: "setTimeout is not a function".

Demo: https://stately.ai/viz?id=a52dd233-64c9-4476-a869-cd6b58b10a12#

### PR DESCRIPTION
**Pull request by mattpocock**
Fixes #205

### CODE REVIEW

File: src/parseMachine.ts [hunk 1]
```diff
@@ -54,6 +54,10 @@ export function parseMachines(sourceJs: string): Array<StateNode> {
       log: console.log,
       warn: console.warn,
     },
+    setInterval,
+    setTimeout,
+    clearTimeout,
+    clearInterval,
```
Comment (Andarist) [inline src/parseMachine.ts:60]: Have you tested this locally though? :troll:

Comment (mattpocock) [reply src/parseMachine.ts:60]: Ooo, good point

Comment (mattpocock) [reply src/parseMachine.ts]: @Andarist Ready for re-review!

File: src/parseMachine.ts [hunk 2]
```diff
@@ -6,6 +6,29 @@ import realmsShim from 'realms-shim';
 
 const realm = realmsShim.makeRootRealm();
 
+/**
+ * Wraps a callback in a function to prevent
+ * accessing 'this'
+ */
```
Comment (Andarist) [inline src/parseMachine.ts:12]: This basically repeats the name of the function - do we need comments like this? 🤔

File: src/parseMachine.ts [hunk 3]
```diff
@@ -6,6 +6,29 @@ import realmsShim from 'realms-shim';
...
+const wrapCallbackToPreventThis = (callback: () => void) => () => {
+  return callback();
+};
```
Comment (Andarist) [inline src/parseMachine.ts:15]: Something like this should improve the compatibility with native timers:
```suggestion
const wrapCallbackToPreventThis = (callback: (...args: any[]) => void) => (...args: any[]) => {
  return callback(...args);
};
```

`setTimeout` and `setInterval` forward arguments to the inner callback, so this works:
```js
setTimeout((a, b) => {}, 0, 'a', 'b')
```

File: src/parseMachine.ts (hunk 2 above)
Comment (mattpocock) [reply src/parseMachine.ts:12]: Sure, I can ditch if you like

File: src/parseMachine.ts (hunk 3 above)
Comment (mattpocock) [reply src/parseMachine.ts:15]: Mmm, good catch

**Review by davidkpiano — APPROVED** [review]

**Review by Andarist — APPROVED** [review]

**Comment by karlguillotte** [comment]
Are you expecting to deploy that fix to the live Viz soon? I am prototyping a machine with the Viz and "setTimeout" is a useful function to have for testing. 

Thanks!

**Pull request merged**